### PR TITLE
[DOCU-2365] Konnect Custom plugins

### PR DESCRIPTION
**Pull request by Guaris**
### Summary
Clarifies limitations around custom plugins

### Reason
[DOCU-2365](https://konghq.atlassian.net/browse/DOCU-2365)

### Testing
coming soon

### CODE REVIEW

File: app/konnect/servicehub/plugins.md [hunk 1]
```diff
@@ -50,20 +51,19 @@ supports.
...
-* Custom plugin database tables: No `dao.lua` file
-* Custom function validations: No function definitions in `schema.lua`
-* Code that runs on the control plane in the plugin handler:
-  * No `init_worker` callback
-  * No Lua code outside of the top-level functions
-* Third-party library dependencies: No `require()` calls to modules that are
-not bundled by default with {{site.konnect_product_name}}
+* Admin API extensions must not contain an `api.lua` file. 
+* Custom plugin database tables must not contain a `dao.lua` file.
+* Custom function validations:
```
Comment (cloudjumpercat) [inline app/konnect/servicehub/plugins.md:58]: For this bullet and the other bullet with subbullets ("Third-party library dependencies:"), I might combine this into one bullet point like "Custom function validations must be written in Lua and self-contain within the schema." But I'm not sure if you nest bullet points in other parts of the docs.

File: app/konnect/servicehub/plugins.md [hunk 2]
```diff
@@ -50,20 +51,19 @@ supports.
...
+  - Must be written in Lua.
+  - Must be self-contained within the schema.
+* Third-party library dependencies: 
+  - Plugins that require third-party libraries can be installed within the handler portion of the plugin code once the plugin is installed on the {{site.base_gateway}} data plane.
```
Comment (cloudjumpercat) [inline app/konnect/servicehub/plugins.md:62]: If you go with my suggestion above, this bullet can probably stand on it's own and wouldn't need a revision.

File: app/konnect/servicehub/plugins.md [hunk 3]
```diff
@@ -50,20 +51,19 @@ supports.
...
+  - The `schema.lua` file must not contain any calls to the `require()` function. 
 
 If your plugin meets these requirements and you want to use it in
 {{site.konnect_saas}}, contact [Kong Support](https://support.konghq.com/).
+
+{:.important}
+> Custom plugin can't be added directly though the {{site.konnect_saas}} application.
```
Comment (cloudjumpercat) [inline app/konnect/servicehub/plugins.md:69]:
```suggestion
> Custom plugins can't be added directly through the {{site.konnect_saas}} application.
```

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]

File: app/konnect/servicehub/plugins.md (hunk 1 above)
Comment (Guaris) [reply app/konnect/servicehub/plugins.md]: Great point.

File: app/konnect/servicehub/plugins.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Manage Plugins
 no_version: true
+content-type: reference
```
Comment (lena-larionova) [inline app/konnect/servicehub/plugins.md:4]:
```suggestion
content_type: reference
```

File: app/konnect/servicehub/plugins.md [hunk 5]
```diff
@@ -50,20 +51,16 @@ supports.
...
-* Code that runs on the control plane in the plugin handler:
-  * No `init_worker` callback
-  * No Lua code outside of the top-level functions
-* Third-party library dependencies: No `require()` calls to modules that are
-not bundled by default with {{site.konnect_product_name}}
+* Admin API extensions must not contain an `api.lua` file. 
+* Custom plugin database tables must not contain a `dao.lua` file.
+* Custom validation functions must be written in Lua and be self-contained within the schema.
+* The `schema.lua` file must not contain any calls to the `require()` function. 
+* Plugins that require third-party libraries can be installed within the handler portion of the plugin code once the plugin is installed on the {{site.base_gateway}} data plane.
```
[the remark this answers is not in the document]
Comment (Guaris) [reply app/konnect/servicehub/plugins.md]: I edited this for clarification.

**Review by lena-larionova — APPROVED** [review]

**Review by cloudjumpercat — APPROVED** [review]

**Pull request merged**